packages/mui-base/src/Tabs/Tab/Tab.types.ts

```diff
@@ -1,17 +1,35 @@
 import * as React from 'react';
-import { UseButtonRootSlotProps } from '../useButton';
+import { BaseUIComponentProps } from '../../utils/BaseUI.types';
+import { TabsOrientation } from '../Root/TabsRoot.types';
+
+export type TabOwnerState = {
+  disabled: boolean;
+  selected: boolean;
+  orientation: TabsOrientation;
+};
+
+export interface TabProps extends Omit<BaseUIComponentProps<'button', TabOwnerState>, 'onChange'> {
+  /**
+   * You can provide your own value. Otherwise, it falls back to the child position index.
+   */
+  value?: any;
+  /**
+   * Callback invoked when new value is being set.
+   */
+  onChange?: (event: React.SyntheticEvent, value: any) => void;
```

Comment:
As discussed, it should be better to prefer different prop names like `onValueChange` that passes in the `value` first.

Reply:
https://github.com/mui/base-ui/pull/245/commits/3a08692f64bad96bb70c76af6f12f18b858abee9